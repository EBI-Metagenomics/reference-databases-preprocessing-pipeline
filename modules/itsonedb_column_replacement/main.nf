
process ITSONEDB_COLUMN_REPLACEMENT {

    label 'light'
    publishDir "${params.outdir}/${label}/${version}/", pattern : "*-tax.txt", mode: 'copy'

    input:
    path uplift
    path taxid
    path tax_header
    val version
    val label

    output:
    path("*-tax.txt"), emit: tax
    path("itsonedb.final.taxid"), emit: taxid

    """
    awk 'BEGIN {FS=OFS="\t"}NR == FNR {a[FNR] = \$B;next}{\$A = a[FNR];print \$0}' B=1 A=1 $uplift $taxid > itsonedb.final.taxid
    grep -v ";p__;" $uplift > uplift.final.filt.txt
    cat $tax_header uplift.final.filt.txt > ${label}-tax.txt
    """

}