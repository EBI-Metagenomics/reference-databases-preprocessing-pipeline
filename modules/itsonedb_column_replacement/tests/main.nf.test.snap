{
    "taxid": {
        "content": [
            [
                "itsonedb.final.taxid:md5,2e61109b60b605dae074b1aee0c54d9c"
            ]
        ],
        "timestamp": "2023-12-12T16:12:00.701436373"
    },
    "tax": {
        "content": [
            [
                "ITSone-tax.txt:md5,96b014a28b10d6ae5a5f69a00fb94c61"
            ]
        ],
        "timestamp": "2023-12-12T16:12:00.652849734"
    }
}